Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -509,8 +509,11 @@ AccessibilityRole AccessibilityNodeObject::determineAccessibilityRoleFromNode(Tr
 
     // There should only be one banner/contentInfo per page. If header/footer are being used within an article or section then it should not be exposed as whole page's banner/contentInfo.
     // https://w3c.github.io/html-aam/#el-header
-    if (node->hasTagName(headerTag) && !isDescendantOfElementType({ articleTag, asideTag, navTag, sectionTag }))
-        return AccessibilityRole::LandmarkBanner;
+    if (node->hasTagName(headerTag)) {
+        if (!isDescendantOfElementType({ articleTag, asideTag, mainTag, navTag, sectionTag }))
```

Comment:
I'm confused why `header` only checks `articleTag, asideTag, mainTag, navTag, sectionTag`, while footer checks `articleTag, asideTag, navTag, sectionTag, mainTag, blockquoteTag, detailsTag, dialogTag, fieldsetTag, figureTag, tdTag`. Should these be the same?

Reply:
Yes, they should check the same element types. I can fix this in a separate patch.